jormungandr/src/explorer/graphql/mod.rs

```diff
@@ -585,12 +585,63 @@ impl OwnerStakeDelegation {
     }
 }
 
+/// Retirement info for a pool
+struct PoolRetirement {
+    pool_retirement: certificate::PoolRetirement,
+}
+
+impl From<certificate::PoolRetirement> for PoolRetirement {
+    fn from(pool_retirement: certificate::PoolRetirement) -> PoolRetirement {
+        PoolRetirement { pool_retirement }
+    }
+}
+
+#[juniper::object(
+    Context = Context,
+)]
+impl PoolRetirement {
+    pub fn pool_id(&self) -> PoolId {
+        PoolId(format!("{}", self.pool_retirement.pool_id))
+    }
+
+    pub fn retirement_time(&self) -> TimeOffsetSeconds {
+        self.pool_retirement.retirement_time.into()
+    }
+}
+
+struct PoolUpdate {
+    pool_update: certificate::PoolUpdate,
+}
+
+impl From<certificate::PoolUpdate> for PoolUpdate {
+    fn from(pool_update: certificate::PoolUpdate) -> PoolUpdate {
+        PoolUpdate { pool_update }
+    }
+}
+
+#[juniper::object(
+    Context = Context,
+)]
+impl PoolUpdate {
+    pub fn pool_id(&self) -> PoolId {
+        PoolId(format!("{}", self.pool_update.pool_id))
+    }
+
+    pub fn start_validity(&self) -> TimeOffsetSeconds {
+        self.pool_update.start_validity.into()
+    }
+
+    // TODO: Previous keys?
+    // TODO: Updated keys?
+}
+
 // TODO can we use jormungandr-lib Certificate ?
```

Comment:
@vincenthz I can't implement juniper's `GraphQLType` for a type in a different crate. Maybe I could try with a newtype